TLM/TLM/Manager/Impl/TrafficLightManager.cs

```diff
@@ -127,6 +129,16 @@ public class TrafficLightManager
             return SetTrafficLight(nodeId, !HasTrafficLight(nodeId, ref node), ref node, out reason);
         }
 
+        public bool CanToggleTL(ushort nodeId) {
```

Comment:
I cannot call it `CanToggleTrafficLight` because overloading that method will break NCR patch.

Reply:
@Elesbaan70  I think using `ToggleTrafficLight` allows more room for future change than using `SetTrafficLight` because this way I don't have to deal with the ambiguity when user sets traffic light to true and there is already a timed simulation.

Reply:
> I cannot call it `CanToggleTrafficLight` because overloading that method will break NCR patch.

You can implement it privately:

```cs
class TrafficLightManager : ITrafficLightManager {

    bool ITrafficLightManager.CanToggleTrafficLight(ushort nodeId) {
        // etc...
    }
}
```

If you do this, the method is only accessible through the `ITrafficLightManager` interface, not through `TrafficLightManager` directly. It has an encoded name that cannot be referenced in code and won't conflict with the existing `CanToggleTrafficLight` method in reflection.